code/src/reducers/tasks.js

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  items: []
+}
+
+export const tasks = createSlice({
+  name: 'tasks',
+  initialState,
+  reducers: {
+    setUpStore: (state, action) => {
+      state.items = action.payload;
+    },
+    addTask: (state, action) => {
+      state.items = [action.payload, ...state.items]
+      // This makes the array immutable (the spread operator)
+      localStorage.setItem('taskList', JSON.stringify(state.items))
+    },
+    deleteAllTasks: (state) => {
+      state.items = [];
+      localStorage.setItem('taskList', JSON.stringify(state.items))
+    },
+    deleteSingleTask: (state, action) => {
+      const id = action.payload; // Destructuring the object in the items array
+      console.log('action.payload', action.payload)
+
+      const copyOfTaskArrayFromStore = state.items;
+      const condition = (element) => element.id === id;
+      const foundIndex = copyOfTaskArrayFromStore.findIndex(condition);
+      copyOfTaskArrayFromStore.splice(foundIndex, 1);
```

Comment:
You could use Immer instead to help prevent hard-to-debug issues related to mutations.